3-async-await/solution.js

```diff
@@ -21,16 +23,46 @@ the format is up to you
 8. as an extra challenge add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
+async function solution() {
+  // YOUR SOLUTION GOES HERE
+  // You generate your id value here
+  const id = Number(Date.now().toString().slice(-2));
+  // You use Promise.all here
+  const data = Promise.all([products(id), price(id)])
+    .then(
+      (res) =>
+        `{
+    id: ${id},\n    product: ${res[0]},
+    price: ${res[1]},\n}`
+    )
+    .catch((error) => {
+      throw new Error(error.message);
+    });
 
-    // You generate your id value here
+  try {
+    const result = await data;
+    console.log(result);
+  } catch (error) {
+    console.log("Error: " + error.message);
+  }
 
-    // You use Promise.all here
+  // You use Promise.allSettled here
+  const status = await Promise.allSettled([products(id), price(id)]);
```

Comment:
add a `try {} catch {}` to this one too

Reply:
It is necessary? because Promise.allSettled, returns the data of the promises even if some have been rejected